napi-inl.h

```diff
@@ -1791,9 +1834,8 @@ inline ClassPropertyDescriptor<T> ObjectWrap<T>::InstanceMethod(
     InstanceMethodCallback method,
     napi_property_attributes attributes,
     void* data) {
-  CallbackData* callbackData = new CallbackData({}); // TODO: Delete when the class is destroyed
-  callbackData->instanceMethodCallback = method;
-  callbackData->data = data;
+  // TODO: Delete when the class is destroyed
+  InstanceMethodCallbackData* callbackData = new InstanceMethodCallbackData({ method, data });
```

Comment:
Anecdotal: I use `auto` pretty heavily and I consider it idiomatic for the node code base (but I introduced most uses so of course I would say that.)

> iirc even Nan and Node core don’t follow the same rules.

Nan is still C++03 (I think) because of v0.10 and v0.12 compatibility.